harness/python/feature.py

```diff
@@ -57,6 +59,7 @@ class Feature:
     expect_run_result: Optional[Any]
     start: Optional[Callable[[Runner], Awaitable[WorkflowHandle]]]
     check_result: Optional[Callable[[Runner, WorkflowHandle], Awaitable[None]]]
+    worker_starter: Optional[Callable[[Runner], Awaitable[None]]]
```

Comment:
This looks like more than just `worker_starter`, it looks like it is also supposed to run the workflow and check results. 

If you just need to be in control of shutting down the worker manually, maybe we should just change this to:

```python
self.start_worker()
try:
  # Rest of stuff
finally:
  self.stop_worker()
```

And use `worker.run()` and `worker.stop()` and stop using `async with`. And `stop_worker` can unset the worker so it knows not to do it if already done.